docs/source/development/design/core.md

```diff
@@ -46,3 +46,130 @@ The system should have:
 - A `config_loader` function to read a particular file, optionally validating it
   against a matching config template.
 - A central `Config` class, which can be built up using `ConfigLoader`.
+
+## Data
+
+We need a system for providing forcing data to the simulation. Although some forcing
+variables are likely to be module specific, it seems better to avoid having arbitrary
+locations and collect everything using configuration of `core.data`.
+
+For the moment, let's assume a common NetCDF format for data inputs.
+
+### Spatial indices
+
+Data is (always?) associated with a grid cell, so the system needs to be able to match
+data values to the cell id of grid cells. The following spatial indexing options seem
+useful (and all of these could add a time dimension):
+
+#### Two dimensial spatial indexing
+
+- Simple indices (`idx_n` and `idx_y`) giving the index of a square grid. The data
+  should match the configured grid size and only really works for square grids (it could
+  just about work for a hex grid with alternate offsets!).
+- Standard `x` and `y` coordinates (or `easting` and `northing`). This would require
+  matching the coordinates of the NetCDF data to the grid definition (origin and
+  resolution) as well as the grid size. This again only really works for square grids.
+
+#### One dimensional spatial indexing
+
+- Simply using a `cell_id` dimension in the NetCDF file to match data to grid cells.
+  This is entirely agnostic about the grid shape - users just need to provide a
+  dimension that covers all of the configure cell ids.
+
+- A **mapping** of particular attributes to sets of cells. The obvious use case here is
+  habitat type style data - for example different soil bulk densities or plant
+  communities. To unpack that a bit:
+
+### Defining mappings
+
+A user can optionally configure mappings, which are loaded and validated before any
+other data. These use one of the first three approaches above, which index _individual_
```

Comment:
This is a bit unclear. You presented two approaches (2d or 1d), could you please clarify?

Reply:
Yup - basically you can't use a mapping to define a mapping. You can _only_ use 2d, or a 1d set of cell_ids.

Reply:
ok, then maybe change it to 'one of the ### two approaches above' ?